File: backend/Models/Job.cs
namespace VisionPaint.Models;

public class Job
{
    public int Id { get; set; }
    public int CompanyId { get; set; } = 1;
    public int? ClientId { get; set; }
    public int? CreatedByPersonId { get; set; }
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string Status { get; set; } = "scheduled";
    public string Priority { get; set; } = "normal";
    public string? AddressLine1 { get; set; }
    public string? AddressLine2 { get; set; }
    public string? City { get; set; }
    public string? StateRegion { get; set; }
    public string? PostalCode { get; set; }
    public string? CountryCode { get; set; }
    public DateTime? ScheduledStartAt { get; set; }
    public DateTime? ScheduledEndAt { get; set; }
    public DateTime? DueDate { get; set; }
    public DateTime? StartedAt { get; set; }
    public DateTime? CompletedAt { get; set; }
    public DateTime? ClosedAt { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

File: frontend/src/types/job.ts
export type JobStatus = "scheduled" | "in_progress" | "completed" | "cancelled";
export type JobPriority = "low" | "normal" | "high" | "urgent";

export type Job = {
  id: number;
  companyId: number;
  clientId: number | null;
  createdByPersonId: number | null;
  title: string;
  description: string | null;
  status: string;
  priority: string;
  addressLine1: string | null;
  addressLine2: string | null;
  city: string | null;
  stateRegion: string | null;
  postalCode: string | null;
  countryCode: string | null;
  scheduledStartAt: string | null;
  scheduledEndAt: string | null;
  dueAt: string | null;
  startedAt: string | null;
  completedAt: string | null;
  closedAt: string | null;
  createdAt: string;
  updatedAt: string;
  photoCount?: number;
};

export type JobInput = {
  clientId?: number | null;
  title: string;
  description?: string | null;
  status?: string;
  priority?: string;
  addressLine1?: string | null;
  addressLine2?: string | null;
  city?: string | null;
  stateRegion?: string | null;
  postalCode?: string | null;
  countryCode?: string | null;
  scheduledStartAt?: string | null;
  scheduledEndAt?: string | null;
  dueAt?: string | null;
};

export type PhotoKind = "before" | "after" | "progress";

export type JobAssignment = {
  personId: number;
  name: string;
  assignmentRole: string;
  assignedAt: string;
};

export type JobDetail = Job & {
  assignments?: JobAssignment[];
};

export type JobPhoto = {
  id: number;
  jobId: number;
  photoKind: PhotoKind;
  caption: string | null;
  takenAt: string;
  uploadedByName: string;
  url: string;
};

export type JobTimeSummary = {
  totalMinutes: number;
  activeMinutes: number;
  byPerson: { personId: number; name: string; minutes: number; inProgress: boolean }[];
};

export type JobStatusHistoryEntry = {
  fromStatus: string | null;
  toStatus: string;
  changedAt: string;
  changedByName: string;
  reason: string | null;
};
